cli.py

```diff
@@ -1,24 +1,82 @@
-import time
-from typing import List, Tuple, Dict
+from files import write_to_file, write_to_file_option, write_interactive_chat_to_file
+from response import clean_chunk, animate_output, clean_response
+from prompts import get_system_prompt, get_user_prompt
+from typing import List
 from models.gpt import GPT
 from tokenizer import Tokenizer
 from models.model_factory import get_valid_models, get_ai_model
 
-# TODO: historico -> salva o historico da conversa atual num txt da vida, e depois le o txt e passa como parametro para o GPT
-
-
 def cli():
     print("=================================================")
     print("Welcome to GPT CLI")
     print("=================================================")
 
-    model_class = choose_model()
-    model = model_class("", [])
-    VARIATIONS = model.get_model_variations()
-    print("Choose your model: ")
+    print("Choose your usage mode")
+    usage_mode = ask_usage_mode()
 
-    model_variation = choose_model_variation(VARIATIONS)
-    model.set_model(model_variation)
+    if usage_mode == 1:
+        one_prompt_mode()
+    else:
+        interactive_conversation_mode()
+
+
+def interactive_conversation_mode():
+    model = prepare_model()
+    print("=================================================")
+    model.set_messages([])
+    system_prompt = get_system_prompt()
+    model.messages.append(system_prompt)
+    print("=================================================")
+    print("YOU'RE NOW STARTING INTERACTIVE CONVERSATION MODE. TYPE 'exit' TO QUIT THE PROGRAM.")
+    print("=================================================")
+
+    tokenizer = None
+    total_cost = 0
+    if isinstance(model, GPT):
+        tokenizer = Tokenizer(model)
+
+    while True:
+        user_prompt = input("You: ")
+        print()
+
+        if user_prompt.lower() == "exit":
+            break
+
+        model.messages.append({"role": "user", "content": user_prompt})
+        stream = model.request(stream=True)
+
+        response = ""
+
+        print("GPT: ", end='')
```

Comment:
Not always GPT, we can use model name to do this. Or the class name. But we can have a "name" attrib on the class for this

Reply:
Thats a good solution. I left it as "Assistant" for now (same name that appears on the exported log).